packages/react-ai/src/components/AIConversation/views/Controls/InputControl.tsx

```diff
@@ -0,0 +1,198 @@
+      </g>
+      <defs>
+        <clipPath id="clip0_387_1426">
+          <rect width="100%" height="100%" />
+        </clipPath>
+      </defs>
+    </svg>
+  ),
+  stroke: 'white',
+  ...baseIconProps,
+});
+
+const SendIcon = withBaseElementProps(Icon, sendIconProps);
+
+const SendButtonBase = withBaseElementProps(Button, {
+  'aria-label': 'Send message',
+  className: `${INPUT_BLOCK}__button ${INPUT_BLOCK}__button--send`,
+  disabled: false,
+});
+
+const SendButton: typeof SendButtonBase = React.forwardRef(function SendButton(
+  { disabled, onClick, ...rest },
+  ref
+) {
+  // TODO should come from context
+  const isWaitingForResponse = false;
+  // TODO send message on click
+  return (
+    <SendButtonBase
+      disabled={isWaitingForResponse}
+      onClick={() => {}}
+      ref={ref}
+      {...rest}
+    />
+  );
+});
+
+const TextInputBase = withBaseElementProps(TextArea, {
+  className: `${INPUT_BLOCK}__input`,
+  placeholder: 'Message Raven',
+  id: `${INPUT_BLOCK}-text-input`,
+});
+
+const TextInput: typeof TextInputBase = React.forwardRef(function TextInput(
+  { onChange, placeholder, ...rest },
+  ref
+) {
+  const [input, setInput] = React.useState('');
+
+  // TODO should come from context or prop
+  const isFirstMessage = false;
+
+  React.useEffect(() => {
+    const textarea = document.getElementById(`${INPUT_BLOCK}-text-input`);
+
+    const handleResize = () => {
+      if (textarea) {
+        textarea.style.height = 'auto';
+        textarea.style.height = `${textarea.scrollHeight}px`;
+      }
+    };
+
+    if (textarea && textarea instanceof HTMLTextAreaElement) {
+      textarea.addEventListener('input', handleResize);
+      handleResize();
+    }
+
+    return () => {
+      if (textarea) {
+        textarea.removeEventListener('input', handleResize);
+      }
+    };
+  }, [input]);
+
+  return (
+    <TextInputBase
+      data-testid="text-input"
+      onChange={(e) => setInput(e.target.value)}
+      {...(isFirstMessage ? { placeholder: 'Ask anything...' } : {})}
+      {...rest}
+      ref={ref}
+    />
+  );
+});
+
+const Container = withBaseElementProps(View, {
+  className: `${INPUT_BLOCK}__container`,
+});
+
+export const InputControl: InputControl = () => {
+  const hiddenInput = React.useRef<HTMLInputElement>(null);
+
+  function handleClick() {
+    if (hiddenInput.current) {
+      hiddenInput.current.click();
+      hiddenInput.current.value = '';
+    }
+  }
+
+  return (
+    <Container>
+      <AttachFileContainer>
+        <AttachFileButton onClick={handleClick}>
+          <AttachFileIcon />
+        </AttachFileButton>
+        <VisuallyHidden>
+          <input
+            data-testid="hidden-file-input"
+            // TODO follow up about what file types are accepted
+            accept="image/*"
+            className={`${INPUT_BLOCK}__attach-file-input`}
+            type="file"
+            ref={hiddenInput}
+          />
+        </VisuallyHidden>
+      </AttachFileContainer>
```

Comment:
This should be its own "control block" component, e.g. `AttachFile`

Reply:
Moved this to its own control exposing just the Button and Icon on its interface